src/kernel/src/signal/set.rs

```diff
@@ -12,32 +11,33 @@ pub struct SignalSet {
 
 impl SignalSet {
     /// An implementation of `SIGISMEMBER`.
-    pub fn contains(&self, sig: NonZeroI32) -> bool {
+    pub fn contains(&self, sig: Signal) -> bool {
         (self.bits[Self::word(sig)] & Self::bit(sig)) != 0
     }
 
     /// An implementation of `SIGADDSET`.
-    pub fn add(&mut self, sig: NonZeroI32) {
+    pub fn add(&mut self, sig: Signal) {
         self.bits[Self::word(sig)] |= Self::bit(sig);
     }
 
     /// An implementation of `SIGDELSET`.
-    pub fn remove(&mut self, sig: NonZeroI32) {
+    pub fn remove(&mut self, sig: Signal) {
         self.bits[Self::word(sig)] &= !Self::bit(sig);
     }
 
     // An implementation of `_SIG_IDX`.
-    fn idx(s: NonZeroI32) -> i32 {
+    fn idx(s: Signal) -> i32 {
         s.get() - 1
     }
 
     /// An implementation of `_SIG_WORD`.
-    fn word(s: NonZeroI32) -> usize {
-        (Self::idx(s) >> 5).try_into().unwrap()
+    fn word(s: Signal) -> usize {
+        // This is safe because `Signal` is guaranteed to be non-negative.
+        unsafe { (Self::idx(s) >> 5).try_into().unwrap_unchecked() }
```

Comment:
This is quite dangerous if someone change `idx` to return a negative value and forget to update this one.